compiler/next/include/chpl/AST/CallExpr.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_AST_CALLEXPR_H
+#define CHPL_AST_CALLEXPR_H
+
+#include "chpl/AST/Expr.h"
+
+#include <vector>
+
+namespace chpl {
+namespace ast {
+
+/**
+  This abstract class represents something call-like
+ */
+class CallExpr : public Expr {
+ public:
+  ~CallExpr() override = 0;
+
+  // note: the reason for the +/- 1 below is that the
+  // 0'th child is the called expression, which does
+  // not count as an "actual".
+
+  int numActuals() const {
+    return this->numChildren() - 1;
+  }
+  const Expr* actual(int i) const {
+    return this->getChild(i+1);
+  }
```

Comment:
This is supposed to be `this->child(i+1)`, right?

Are we sure to number the actuals from 0?  "call->actual(1) gives us the second actual" does not sound good to me.

How much performance gain would we get by specializing for the common case of low i, ex. by introducing actual1(), actual2() ?

Reply:
> Are we sure to number the actuals from 0?

We're a 0-based language now, right? :)

I wouldn't expect much performance gain from specializing these.